aiida_vasp/workflows/relax.py

```diff
@@ -0,0 +1,298 @@
+"""Structure relaxation workchain for VASP."""
+import numpy
+from aiida.common.extendeddicts import AttributeDict
+from aiida.work import WorkChain
+from aiida.work.workchain import append_, while_, ToContext
+
+from aiida_vasp.utils.aiida_utils import get_data_class, get_data_node
+from aiida_vasp.utils.vasp.isif import IsifStressFlags, Isif
+from aiida_vasp.utils.vasp.ibrion import IbrionFlags, Ibrion
+from aiida_vasp.utils.vasp.encut import EncutFlags, Encut
+from aiida_vasp.utils.vasp.nsw import Nsw
+from aiida_vasp.workflows.base import VaspBaseWf
+from aiida_vasp.workflows.restart import prepare_process_inputs, UnexpectedCalculationFailure
+
+RELAXATION_INCAR_TEMPLATE = AttributeDict({
+    "ismear": -1,
+    "sigma": 0.2,
+    "algo": 'NORMAL',
+    "ediff": 1E-6,
+    "prec": 'Normal',
+    "ibrion": 2,
+    "nsw": 200,
+    "nelmin": 4,
+    "ediffg": -0.01,
+    "isif": 4
+})
+
+
+def relax_parameters(pos, shape, volume):
+    """Set IBRION and ISIF according to the choice of what should be relaxed."""
+    if not (pos or volume or shape):
+        raise ValueError('relaxation workflow requires at least one of "positions", "cell shape", "cell volume" to be alowed to change.')
+    if not pos:
+        ibrion = Ibrion(ion_updates=IbrionFlags.NO_UPDATE)
+    else:
+        ibrion = Ibrion(ion_updates=IbrionFlags.IONIC_RELAXATION_CG)
+    isif = Isif(calculate_stress=IsifStressFlags.FULL, vary_positions=pos, vary_cell_shape=shape, vary_cell_volume=volume)
+    nsw = Nsw(value=200, ibrion=ibrion)
+    return ibrion, isif, nsw
+
+
+def encut_factor(volume):
+    if volume:
+        return 1.3
+    return 1
+
+
+def clean_incar_overrides(inputs):
```

Comment:
We might consider how strict we are going to be, or at least give options to turn these things off. Quite frequently we test VASP code and supply silly parameters that would, if used with this plugin not execute.

Reply:
I agree in general, however this is not the workchain for doing tests, this is only for production relaxations, If you already know you want silly parameters, you will use the `base` or write your ow test workchain more likely.

Consider we allowed to override these, and then someone queries for `relax` workchains with `relax.positions==True`. There would be no way of knowing that the workchains they get back actually do relax positions, it might be overridden. There is no way out of this, since the workchain can not change it's inputs.

Reply:
This sounds a bit more work than it needs to be from the VASP user side. Often the test might not be very experimental, it can be minor tweaks, such that one would like to run the whole chain again etc. Back to the INCAR above, when we do b) this is fine. But that has to be accepted from the top through the workchains. It could in practice simply mean running with a) then discovering..."ah this issue again", recompiling VASP with a specific method and parameters and execute the same thing as option b). I think this would give us the least headache with customization and overrides as there are no middle ground. In any case we can fill out that middle ground later if the users find this useful.